docs/src/pages/theming.mdx

```diff
@@ -0,0 +1,430 @@
+import { Callout } from "nextra-theme-docs";
+
+# Theming
+
+## Components anatomy
+
+### UploadButton
+
+Simplified component structure:
+
+```jsx
+<div data-state={getUtState()} className={className}>
+  <label data-state={getUtState()} data-ut-element="button">
+    <input />
+    /button content goes here/
+  </label>
+  <div data-state={getUtState()} data-ut-element="allowed-content">
+    /allowed content text goes here/
+  </div>
+</div>
+```
+
+<Callout type="info">
+  UploadButton consists of three themable elements: container, button and
+  allowed content.
+</Callout>
+
+<Callout type="warning">
+  Note: UploadButton's button element is defined using `label`
+</Callout>
+
+import Image from "next/image";
+
+<Image
+  src="/upload-button-anatomy.png"
+  alt="Upload button anatomy"
+  width="1546"
+  height="998"
+  style={{ marginTop: "1rem", borderRadius: "0.5rem" }}
+/>
+
+### UploadDropzone
+
+Simplified component structure:
+
+```jsx
+<div className={className} data-state={getUtState()}>
+  <svg data-ut-element="upload-icon" data-state={getUtState()}>
+    ...
+  </svg>
+  <label data-ut-element="label" data-state={getUtState()}>
+    /label content goes here/
+    <input disabled={!ready} />
+  </label>
+  <div data-ut-element="allowed-content" data-state={getUtState()}>
+    /allowed content goes here/
+  </div>
+  <button
+    data-ut-element="button"
+    data-state={getUtState()}
+    disabled={isUploading}
+  >
+    /button content goes here/
+  </button>
+</div>
+```
+
+<Callout type="info">
+  UploadDropzone consists of five themable elements: container, upload icon,
+  label, button and allowed content.
+</Callout>
+
+<Callout type="warning">
```

Comment:
It is not quite clear to me what this means, maybe we should refactor the components to make them have the same anatomy?

Reply:
Regarding what that means: Visually buttons in both UploadButton and UploadDropzone are the similar. On the markup level they are defined using different tags. In case of `UploadButton` it's `label` because it's wrapping hidden file input. To make styling through data attributes(and custom data classes from TW specifically) work in the same way among both `UploadButton` and `UploadDropzone` I introduced `data-ut-element` data attribute. So if `UploadButton` has some styles applied to button element, for example, `ut-button:text-cyan-500`, it can be just copy pasted to `UploadDropzone` and those styles will be applied to the button of `UploadDropzone`. Hope my explanation is clear in this case

Regarding anatomy refactoring: I guess we kind of can do that. We either need to replace `label` with `button` inside of `UploadButton` or `button` with `label` inside of `UploadDropzone`. Maybe there is also another option I'm missing. What do you think?